docs/faq.rst

```diff
@@ -59,3 +59,60 @@ This error can occure within requests' ``session.py`` when called by
 the destructor (``__del__``) of browser. The solution is to
 call :func:`~mechanicalsoup.Browser.close` before the end of life of
 the object.
+
+How do I get debug information/logs
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To unerstand what's going on while running a script, you have two
+options:
+
+* Use :func:`~mechanicalsoup.StatefulBrowser.set_debug` to set the
+  debug level to 1 (show one dot for each page opened, a poor man's
+  progress bar) or 2 (show the URL of each visited page).
+
+* Activate request's logging::
+
+    import requests
+    import logging
+
+    logging.getLogger().setLevel(logging.DEBUG)
+    requests_log = logging.getLogger("requests.packages.urllib3")
+    requests_log.setLevel(logging.DEBUG)
+    requests_log.propagate = True
+
+  This will display a much more verbose output, including HTTP status
+  code for each page visited. Note that unlike MechanicalSoup's
+  logging system, this includes URL returning a redirect (e.g. HTTP
+  301), that are dealt with automatically by requests and not visible
+  to MechanicalSoup.
+
+Should I use Browser or StatefulBrowser?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Short answer: :class:`mechanicalsoup.StatefulBrowser`.
+
+:class:`mechanicalsoup.Browser` is historically the first class that
+was introduced in Mechanicalsoup. Using it is a bit verbose, as the
+caller needs to store the URL of the currently visited page and
+manipulate the current form with a separate
+variable. :class:`mechanicalsoup.StatefulBrowser` is essentially a
+superset of :class:`mechanicalsoup.Browser`, it's the one you should
+use unless you have a good reason to do otherwise.
+
+How does MechanicalSoup compare to the alternatives?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+There are other libraries with the same purpose as MechanicalSoup:
+
+* `mechanize <http://wwwsearch.sourceforge.net/mechanize/>`__, an
+  ancestor of MechanicalSoup (getting its name from the Perl mechanize
+  module). It was a great tool, but doesn't support Python 3, isn't
+  maintained anymore, and its documentation is poor. You shouldn't use
+  it anymore.
```

Comment:
As of this year, there is a new maintainer of Mechanize. There are extensive docs now (https://mechanize.readthedocs.io/en/latest/), but he has no intention of porting to python3 (https://github.com/python-mechanize/mechanize/issues/9). So I would agree that it shouldn't be used anymore, but the other statements might be a bit harsh.

Reply:
Indeed, thanks for the pointers.

Reply:
Not really true, this was always more of the "patches welcome" variety. And mechanize has finally been updated to support python3 on master.